drivers/SmartThings/matter-lock/src/init.lua

```diff
@@ -113,24 +116,26 @@ end
 
 local function num_pin_users_handler(driver, device, ib, response)
   device:set_field(lock_utils.TOTAL_PIN_USERS, ib.data.value)
-  local creds_per_user = device:get_field(lock_utils.CREDENTIALS_PER_USER)
-  if creds_per_user and creds_per_user > 0 then
-    device:emit_event(capabilities.lockCodes.maxCodes(ib.data.value * creds_per_user, {visibility = {displayed = false}}))
-  end
+  device:emit_event(capabilities.lockCodes.maxCodes(ib.data.value, {visibility = {displayed = false}}))
 end
 
-local function num_creds_per_user_handler(driver, device, ib, response)
-  device:set_field(lock_utils.CREDENTIALS_PER_USER, ib.data.value)
-  local num_pin_users = device:get_field(lock_utils.TOTAL_PIN_USERS)
-  if num_pin_users and num_pin_users > 0 then
-    device:emit_event(capabilities.lockCodes.maxCodes(ib.data.value * num_pin_users, {visibility = {displayed = false}}))
+local function require_remote_pin_handler(driver, device, ib, response)
+  if ib.data.value then
+    --Process after all other info blocks have been dispatched to ensure MaxPINCodeLength has been processed
```

Comment:
does this delay(0) *guarantee* this, or just make it likely?

Reply:
Yes, the device responds to our read of MinPINCodeLength, MaxPINCodeLength, NumberOfPINUsersSupported, and RequirePINForRemoteOperation, all in one interaction response (I think this is guaranteed by the SDK). Our dispatcher processes an entire interaction response as one driver "event", so all of those attribute reports will be processed by handlers before the driver does anything else. Doing the delay(0) queues up a new event which will be processed after all the attribute reports, which is needed since the order of attribute report info blocks is not guaranteed in the response.